pageserver/src/layered_repository/page_versions.rs

```diff
@@ -39,6 +39,10 @@ impl PageVersions {
         }
     }
 
+    pub fn size(&self) -> u64 {
+        self.file.pos
```

Comment:
Please look at `append_or_update_last`. It assumes that  self.file.stream_position() points to the end of file.
I am not using `self.file.stream_position()` because it modifies `self` object and it is inconvenient.

Reply:
For this PR it makes sense.

But I'm curious: If EphemeralFile is supposed to be append-only, why do we implement Seek? We only use seek in append_or_update_last, but that line can be replaced with `self.file.write_all(&[0; 4]);`

Also, in `EphemeralFile::create` we initialize with `pos: 0`. And we don't delete the virtual file if it already exists, so it looks like we'd break the "pos is at the end" invariant on process restart.

Reply:
I think @hlinnaka can better answer on this question, because this is his code.
But my understanding is that EphemeralFile can be used not only in PageVersions and so must no be append-only.
But in PageVersions it is used in append-only manner and it make it possible to use current file position as file size.

Concerning creation of EphemeralFile - we always choose unique name for this file: `ephemeral-{}, l.next_file_id`.
So it should not exists. On pageserver restart all ephemeral files are deleted. But I agree with you, for safety we should open VirtualFile in `EphemeralFile::create` with `create_new(true)` or `create(true).truncate(true)` option. @hlinnaka, what do you think?

Reply:
> But I'm curious: If EphemeralFile is supposed to be append-only, why do we implement Seek?

I tried to make it similar to a regular File, but I guess we didn't need Seek in the end.  Feel free to change it as makes sense!

> Also, in EphemeralFile::create we initialize with pos: 0. And we don't delete the virtual file if it already exists, so it looks like we'd break the "pos is at the end" invariant on process restart.

We delete all ephemeral files at pageserver startup, see https://github.com/zenithdb/zenith/blob/main/pageserver/src/layered_repository.rs#L1169. Could use some comments in `ephemeral_file.rs` to point that out, I guess..

Reply:
> But I agree with you, for safety we should open VirtualFile in EphemeralFile::create with create_new(true) or create(true).truncate(true) option.

Well, the file shouldn't exist in the first place. If it does, something has already gone wrong, so I'm not sure what the best action would be. Plow ahead and overwrite the existing file, or be cautious and error out to prevent further damage? Or issue a warning and then overwrite? I'm not sure.

Reply:
Works for me